packages/vite/src/node/plugins/resolve.ts

```diff
@@ -601,6 +608,14 @@ export function resolvePackageEntry(
 
   entryPoint = entryPoint || data.main || 'index.js'
 
+  // make sure we don't get scripts when looking for sass
+  if (options.mainFields && options.mainFields[0] == 'sass' && options.extensions) {
+    if (!options.extensions.includes(path.extname(entryPoint))) {
+      entryPoint = '';
+      options.skipPackage = true;
+    }
+  }
```

Comment:
**suggestion:** I think this can simplified to:

```suggestion
  if (
    options.mainFields?.[0] === 'sass' &&
    !(options.extensions ?? []).includes(path.extname(entryPoint))
  ) {
      entryPoint = '';
      options.skipPackage = true;
  }
```

- Maybe the formatting is different :thinking:  
- Maybe the `options.extensions ?? []` part can also be written in a way using `?.` :thinking:

Reply:
Thanks, I've changed that. This does introduce a subtle change, but I don't think this situation can actually occur. If a sass file is expected, but `options.extensions` is undefined, then any discovered package.json entry will not be used.

Reply:
Tests passes, so at least these say it is okay ^^'
If not, someone will fix it and provide a test :slightly_smiling_face: